### PR TITLE
Reformatted Musig

### PR DESCRIPTION
**Pull request by elichai**
Rephrased "individual signatures" to "partial signatures".
Changed `sign_musig()` to return just the `s` as an int.
Changed `aggregate_musig_signatures()` to accept a list of *int* s's and the aggregated R,

### CODE REVIEW

File: test_framework/musig.py [hunk 1]
```diff
@@ -52,26 +52,15 @@ def sign_musig(priv_key, k_key, R_musig, P_musig, msg):
     """Construct a musig signature."""
```
Comment (jnewbery) [inline test_framework/musig.py]: Update this comment to say "Construct a MuSig partial signature and return the s value."

File: test_framework/musig.py [hunk 2]
```diff
@@ -52,26 +52,15 @@ def sign_musig(priv_key, k_key, R_musig, P_musig, msg):
...
-    R = SECP256K1.affine(SECP256K1.mul([(SECP256K1_G, k_key.secret)]))
-    e = int.from_bytes(hashlib.sha256(Rm[0].to_bytes(32, 'big') + P_musig.get_bytes() + msg).digest(), 'big') % SECP256K1_ORDER
-    return R[0].to_bytes(32, 'big') + ((k_key.secret + e * priv_key.secret) % SECP256K1_ORDER).to_bytes(32, 'big')
+    assert jacobi_symbol(R_musig.get_y(), SECP256K1_FIELD_SIZE) == 1
+    e = int.from_bytes(hashlib.sha256(R_musig.get_x().to_bytes(32, 'big') + P_musig.get_bytes() + msg).digest(), 'big') % SECP256K1_ORDER
+    return (k_key.secret + e * priv_key.secret) % SECP256K1_ORDER
 
-def aggregate_musig_signatures(sigs):
+def aggregate_musig_signatures(s_list, R_musig):
     """Construct valid Schnorr signature from individually generated musig signatures."""
```
Comment (jnewbery) [inline test_framework/musig.py]: Update comment to say 'partial MuSig signatures'

File: test_framework/musig.py [hunk 3]
```diff
@@ -52,26 +52,15 @@ def sign_musig(priv_key, k_key, R_musig, P_musig, msg):
     """Construct a musig signature."""
     assert priv_key.valid
     assert priv_key.compressed
+    assert P_musig.compressed
     assert len(msg) == 32
     assert k_key is not None and k_key.secret != 0
-    Rm = SECP256K1.affine(R_musig.p)
-    assert jacobi_symbol(Rm[1], SECP256K1_FIELD_SIZE) == 1
-    R = SECP256K1.affine(SECP256K1.mul([(SECP256K1_G, k_key.secret)]))
```
Comment (jnewbery) [inline test_framework/musig.py]: `SECP256K1_G` is no longer used. You can remove the import.

File: test_framework/musig.py [hunk 4]
```diff
@@ -52,26 +52,15 @@ def sign_musig(priv_key, k_key, R_musig, P_musig, msg):
...
-    for sig in sigs:
-        assert len(sig) == 64
-        s_list.append(int.from_bytes(sig[32:], 'big'))
-        R = SECP256K1.lift_x(int.from_bytes(sig[:32], 'big'))
-        if jacobi_symbol(R[1], SECP256K1_FIELD_SIZE) != 1:
-            R = SECP256K1.negate(R)
-        R_list.append(R)
+    assert s_list is not None and all(isinstance(s, int) for s in s_list)
     s_agg = sum(s_list) % SECP256K1_ORDER
-    R_agg = reduce(lambda x, y: SECP256K1.add_mixed(x, y), R_list)
```
Comment (jnewbery) [inline test_framework/musig.py]: `reduce` is no longer used. You can remove the import.

File: 1.2-Introduction-to-Musig.ipynb [hunk 5]
```diff
@@ -141,9 +141,9 @@
     "\n",
     "In this exercise, we'll generate nonces for individual participants, calculate the nonce point commitments, and then generate an aggregate nonce point.\n",
     "\n",
-    "* Use `generate_schnorr_nonce()` to generate a random nonce. The function returns an `ECkey` object which is a valid schnorr nonce (quadratic residue y-value).\n",
-    "* Use `aggregate_schnorr_nonces()` to aggregate those individual nonces. The function takes a list of `ECKey` nonces and returns a the aggregate nonce and a negation flag:\n",
-    "  * the aggregate nonce is an `ECPubKey` object\n",
+    "* Use `generate_schnorr_nonce()` to generate a random nonce. The function returns an `ECkey` object which is a valid schnorr nonce.\n",
+    "* Use `aggregate_schnorr_nonces(nonce_point_list)` to aggregate those individual nonces. The function takes a list of `ECKey` nonces and returns a the aggregate nonce and a negation flag:\n",
+    "  * the aggregate nonce is a `ECPubKey` object\n",
```
Comment (jnewbery) [inline 1.2-Introduction-to-Musig.ipynb]: nit: this should be 'an `ECPubKey`', not 'a `ECPubKey`'

File: 1.2-Introduction-to-Musig.ipynb [hunk 6]
```diff
@@ -206,18 +206,18 @@
...
-    "Use the `sign_musig()` function to create individual signatures. `sign_musig()` takes:\n",
+    "Use the `sign_musig()` function to create partial signatures. `sign_musig()` takes:\n",
     "  - the individual participant's challenge-factor-modified private key (an `ECKey` object)\n",
     "  - the invididual participant's nonce scalar (an `ECKey` object)\n",
     "  - the aggregate nonce point (an `ECPubKey` object)\n",
     "  - the aggregate pubkey (an `ECPubKey` object)\n",
     "  - the message (a 32 byte `bytes` object)\n",
     "\n",
-    "and returns an individual signature (a 64 byte `bytes` object containing `R(x)` and `s`).\n",
+    "and returns a partial signature (a int containing the partial `s` value).\n",
```
Comment (jnewbery) [inline 1.2-Introduction-to-Musig.ipynb]: s/a int/an int/

**Comment by jnewbery** [comment]
I've pushed a commit that updates chapters 2.1 and 2.2 to work with the new functions.

File: 1.2-Introduction-to-Musig.ipynb (hunk 5 above)
Comment (elichai) [reply 1.2-Introduction-to-Musig.ipynb]: really? ha.

File: 1.2-Introduction-to-Musig.ipynb (hunk 6 above)
Comment (elichai) [reply 1.2-Introduction-to-Musig.ipynb]: Done

File: 1.2-Introduction-to-Musig.ipynb (hunk 5 above)
Comment (elichai) [reply 1.2-Introduction-to-Musig.ipynb]: Done

File: test_framework/musig.py (hunk 4 above)
Comment (elichai) [reply test_framework/musig.py]: Done

File: test_framework/musig.py (hunk 3 above)
Comment (elichai) [reply test_framework/musig.py]: Done, thanks!

File: test_framework/musig.py (hunk 2 above)
Comment (elichai) [reply test_framework/musig.py]: Done. thanks.

File: test_framework/musig.py (hunk 1 above)
Comment (elichai) [reply test_framework/musig.py]: Done

File: solutions/1.2-Introduction-to-Musig-Solutions.txt [hunk 7]
```diff
@@ -62,13 +62,13 @@ print("R_agg was negated:", negated)
...
-sig2 = sign_musig(privkey2_c, k2, R_agg, pubkey_musig, msg)
+# Generate partial signatures
+s0 = sign_musig(privkey0_c, k0, R_agg, pubkey_musig, msg)
+s1 = sign_musig(privkey1_c, k1, R_agg, pubkey_musig, msg)
+s2 = sign_musig(privkey2_c, k2, R_agg, pubkey_musig, msg)
 
-# Aggregate signatures
-sig_agg = aggregate_musig_signatures([sig0, sig1, sig2])
+# Aggregate partial signatures
+sig_agg = aggregate_musig_signatures([s0, s1, s2])
```
Comment (jnewbery) [inline solutions/1.2-Introduction-to-Musig-Solutions.txt]: This needs to take `R_agg`:

`sig_agg = aggregate_musig_signatures([s0, s1, s2], R_agg)`

**Comment by jnewbery** [comment]
pushed a commit to fix the aggregate_musig_signatures() call in solutions 1.2.

**Comment by jnewbery** [comment]
squashed commits

**Pull request merged**